lib/improver/metadata/utilities.py

```diff
@@ -81,6 +103,34 @@ def create_new_diagnostic_cube(
     return cube
 
 
+def generate_mandatory_attributes(diagnostic_cubes):
```

Comment:
I don't think this is actually used anywhere

Reply:
Not yet.  It's something I think is needed to support `create_new_diagnostic_cube`, which I'm expecting to use in follow-on tickets to implement cube creation (where mandatory attributes are not inherited automatically from source cubes).  But I'm open to discussing that?

Reply:
See comments on PR #1020 to see envisaged use.

Reply:
Ok, I can see that use case. So the behaviour is currently:
* If all the input cubes used to generate a new cube have the same values for the mandatory attribute then inherit them.
* If the don't match then set to the default.
Do you also plan to have the option to provide new values for the mandatory attributes for plugins that create new cubes? Or only use the two options already outlined?

Reply:
So I think the idea is:
* Use this function to check all the input cubes have the same values for each mandatory attributes. If they don't match, then set to the default.
* Pass these new attributes to the cube creation function.

Is there going to be an option for a user to specify new values for the mandatory attributes too?

Reply:
Yes - via the "optional_attributes" argument to create_new_diagnostic_cube.  This allows the plugin to overwrite default values where the input cubes do not match.

Reply:
But this will not be needed in many cases, as often when we are creating new diagnostics we are using information from the same source model, so all the necessary metadata will agree.

Reply:
Basically, there is a bit of design decision here that is worth getting feedback on.  I could have written `generate_mandatory_attributes` to be called from `create_new_diagnostic_cube`.  This might be easier to use / interpret.  However, there were a couple of drawbacks that I could see:

1. The user might actually want the plugin to set up the mandatory attributes dictionary itself, with plugin-specific values, rather than using the "generate defaults" function.
1. The "generate defaults" function requires passing in the whole list of input diagnostic cubes.  This seemed like overkill for `create_new_diagnostic_cube`, especially if in some cases those cubes wouldn't be processed at all; but also it's actually not appropriate for things like nowcasting, only for new diagnostics which are calculated using several model diagnostics.  So wrapping this into the general utility makes the utility itself less general.

So on balance I decided to have these as two separate functions - a cube creation utility that requires values for the mandatory attributes, and a function that can generate those attributes using a list of input cubes if needed.

Reply:
I was more meaning at the higher level interface in the plugin that is creating the new diagnostic. But I think this is more to do with how this new function is called in the plugins that are building new diagnostic and is not really a question for this PR. So I'm happy with this utility for now.